### ISSUE TITLE
Geosearch filter should sort the geo points for the polygon

### ISSUE THREAD

**Issue by nmorenor** [typesense/typesense#331]
## Description

While drawing a polygon on a google Map, following this example on [stackoverflow](https://stackoverflow.com/questions/14166546/google-maps-drawing-manager-limit-to-1-polygon). After converting the polygon to a typesense filter sometimes I did get the right search results, but other times I just had all the documents on the collection.

## Steps to reproduce

Follow the stackoverflow to build a map drawing a polygon, convert the polygon to a search query like:

This query does not work:

```{"q":"*","query_by":"title,address,listingId","page":1,"filter_by":"location:(20.643101970825, -105.2271453804, 20.647860850082, -105.22630853119, 20.646696243606, -105.22096557083, 20.646897038462, -105.22055787506, 20.646756482091, -105.22010726395, 20.641214441604, -105.22118014755, 20.641234521826, -105.22208136978, 20.642057808619, -105.22328299942)"}```

This query does work:

```"{"q":"*","query_by":"title,address,listingId","page":1,"filter_by":"location:(20.647836009729, -105.22632746033, 20.643036969737, -105.22716430954, 20.642032967319, -105.22321609787, 20.641149439709, -105.2214994841, 20.641591204155, -105.22109178833, 20.646832038996, -105.22008327774)"}"```

A workaround for this to work will be to check the polygon just before sending the search to typesense. In my case on php I just do the following:

```
$polygonSource = $sourceSearch->polygon;
if ($polygonSource[0]->lat < $polygonSource[1]->lat) {
    $polygonSource = array_reverse($polygonSource);
}
```

The issue is the order in which the user drawing on the map draws the polygon, if the user add points on clockwise direction, the search does not filter the results.

## Expected Behavior

Regardless of the point locations on the polygon, the query should return a good result

## Actual Behavior

In counter clock wise search does filter the hits. On clock wise polygon it does not return filtered results.

## Metadata

**Typsense Version**: 0.21.0

**OS**: Running the docker image. Host is OSx 10.15.17

**Comment by jasonbosco** [comment]
Geosearch-ing within a polygon requires the polygon's edges to be specified in an anti-clockwise direction [by design](https://typesense.org/docs/0.21.0/api/documents.html#searching-within-a-geo-polygon):

> The polygon's points must be defined in a counter-clockwise (i.e. anti-clockwise) direction.

Thank you for sharing your solution to sort geo points!

**Comment by kishorenc** [comment]
@nmorenor We should be certainly doing this 🤦 Reopening this issue so that we can fix this for the next release!

**Comment by kishorenc** [comment]
This has been fixed for the next release and can be verified on the `0.22.0.rcs22` Docker image.

**Comment by jasonbosco** [comment]
Released in v0.22
